pymc3/distributions/multivariate.py

```diff
@@ -1925,6 +1924,81 @@ def _distr_parameters_for_repr(self):
         return ["mu"]
 
 
+class CARRV(RandomVariable):
+    name = "car"
+    ndim_supp = 1
+    ndims_params = [1, 2, 0, 0]
+    dtype = "floatX"
+    _print_name = ("CAR", "\\operatorname{CAR}")
+
+    def make_node(self, rng, size, dtype, mu, W, alpha, tau):
+        mu = at.as_tensor_variable(floatX(mu))
+
+        W = aesara.sparse.as_sparse_or_tensor_variable(floatX(W))
+        if not W.ndim == 2:
+            raise ValueError("W must be a matrix (ndim=2).")
+
+        sparse = isinstance(W, aesara.sparse.SparseVariable)
```

Comment:
I am surprised if this works, the inputs are not evaluated inside the `make_node` are they? I think you have to add a Aesara assert instead of a Python exception

Reply:
There is an example here: https://github.com/pymc-devs/pymc3/blob/e8396bf8961500b63221db45b1249b92bf4afdbd/pymc3/distributions/continuous.py#L201

Reply:
@ricardoV94, I added ```Assert``` in these checks.